object_dynamic.go

```diff
@@ -141,7 +141,7 @@ func (r *Runtime) NewDynamicArray(a DynamicArray) *Object {
 		a: a,
 		baseDynamicObject: baseDynamicObject{
```

Comment:
This was just a matter of pulling in only the changes required to get it working since I didn't actually use a cherry-pick. I do actually like this change though so I'll go ahead and pull this over.

I did notice that upstream also missed updating this in the [function above](https://github.com/dop251/goja/blob/master/object.go#L862C13-L862C13). I wonder if this was just an oversight.